client.go

```diff
@@ -123,10 +136,32 @@ func NewClient() *Client {
 	}
 }
 
+// ErrStopRetry is returned by a CheckRetry callback to stop retries, but
+// still have the client return the original error.
+var ErrStopRetry = errors.New("stop retry")
+
 // Do wraps calling an HTTP method with retries.
 func (c *Client) Do(req *Request) (*http.Response, error) {
 	c.Logger.Printf("[DEBUG] %s %s", req.Method, req.URL)
 
+	if c.CheckRetry == nil {
+		// Check the response code. We retry on 500-range responses to allow
+		// the server time to recover, as 500's are typically not permanent
+		// errors and may relate to outages on the server side.
+		c.CheckRetry = func(resp *http.Response, err error) error {
```

Comment:
Let's make this a package-level function and default to it rather than inlining. It would also make sense to do this default assignment in `NewClient()` rather than the request path.